assets/js/main.js

```diff
@@ -520,107 +520,16 @@ function setBackLink(){
- casebookJSON.premises  = subBuilding.substr(subBuilding.split(" ")[0].length, subBuilding.length-1).replace(',','') ;
- }
-
-
- }else{
- casebookJSON.premises  =address.house_name.substr(0,address.house_name.length- house_name[house_name.length-1].length).replace(',','');
-
- }
- }
- else if(house_name[0].toLowerCase()=="flat"  && $.isNumeric(house_name[1].replace(',','') )){
- console.log(1);
- casebookJSON.flatNumber = house_name[1].replace(',','');
- casebookJSON.premises  =address.house_name.substr(house_name[0].length +house_name[1].length+1,address.house_name.length ).replace(',','');
- }
- else if($.isNumeric(house_name[0].split(/[A-Za-z]/)[0])){
- console.log(5);
- casebookJSON.houseNumber = house_name[0];
- casebookJSON.premises  =address.house_name.substr(house_name[0].length+1,address.house_name.length ).replace(',','');
- }else if($.isNumeric(house_name[0].replace("-",""))){
- console.log(6);
- casebookJSON.houseNumber = house_name[0].replace(',','');
- casebookJSON.premises  =address.house_name.substr(house_name[0].length+1,address.house_name.length ).replace(',','');
- }
- else if($.isNumeric(house_name[0].replace(",",""))){
- console.log("OX18 4JU");
- casebookJSON.houseNumber = house_name[0].replace(',','');
- casebookJSON.premises  =address.house_name.substr(house_name[0].length+1,address.house_name.length-house_name[0].length+1 ).replace(',','');
- }
- else if(address.house_name.length>10 ){
- casebookJSON.premises =address.house_name.replace(',','');
- }
- else {
- casebookJSON.premises = address.house_name
- }
-
- console.log(casebookJSON);
- var result = '<h1 class="heading-medium">Address test</h1>';
- result += '<table><tr><th>Casebook Field</th><th>Result</th></tr>';
- result += '<tr><td>Company name</td><td>'+casebookJSON.companyName+'</td></tr>';
- result += '<tr><td>Flat number</td><td>'+casebookJSON.flatNumber+'</td></tr>';
- result += '<tr><td>Premises</td><td>'+casebookJSON.premises+'</td></tr>';
- result += '<tr><td>House number</td><td>'+casebookJSON.houseNumber+'</td></tr>';
- result += '<tr><td>Street</td><td>'+casebookJSON.street+'</td></tr>';
- result += '<tr><td>District</td><td>'+casebookJSON.district+'</td></tr>';
- result += '<tr><td>Town</td><td>'+casebookJSON.town+'</td></tr>';
- result += '<tr><td>Region</td><td>'+casebookJSON.region+'</td></tr>';
- result += '<tr><td>Postcode</td><td>'+casebookJSON.postcode+'</td></tr>';
- result += '<tr><td>Country</td><td>'+casebookJSON.country+'</td></tr>';
- result += '</table>';
- $('#test-address-result').html(result);
- }
- */
+(function hideUploadFileButton() {
+  var fileUploadInput = document.querySelector(".js-file-auto-upload");
+  var fileUploadButton = document.querySelector(".js-trigger-progress-bar");
+
+  if (fileUploadInput) {
+    fileUploadButton.style.display = "none";
+
+    // Automatically click upload button when file is selected
+    fileUploadInput.onchange = () => {
+      fileUploadButton.click();
+    };
+  }
+})();
```

Comment:
This is the main change

Reply:
@RichardBray does this work on a mobile device?  Worth checking the click function particularly - this post suggests it will but worth a check: https://stackoverflow.com/questions/37273142/would-onclick-event-work-on-touch-on-touch-screen-devices.  

Also, is there a fallback when JS is disabled or is it dependent on callign the `hideUploadFileButton()` function?

Reply:
What's the rest of the app like in terms of no JS?

Reply:
> @RichardBray does this work on a mobile device? Worth checking the click function particularly - this post suggests it will but worth a check: https://stackoverflow.com/questions/37273142/would-onclick-event-work-on-touch-on-touch-screen-devices.
> 
> Also, is there a fallback when JS is disabled or is it dependent on callign the `hideUploadFileButton()` function?

Hey @alisalaman-cyb thanks for sharing the link. Luckily it all works on mobile 🎉 

<img width="572" alt="Screenshot 2022-04-19 at 16 41 41" src="https://user-images.githubusercontent.com/1377253/164042914-3c19c644-46ae-47fb-9885-4b28fc366f8b.png">

When JS is disabled there will be an Upload button that the user has to press to upload the file. The `hideUploadFileButton()` hides that button.

<img width="882" alt="Screenshot 2022-04-19 at 16 44 35" src="https://user-images.githubusercontent.com/1377253/164043253-3800f531-9ab2-40b9-8995-31fa76926fa9.png">

Reply:
> What's the rest of the app like in terms of no JS?

Hey @PaulioRandall The whole app works fine if JS is disabled actually. The server side code takes priority and the client side JS is just sprinkled on top only when needed.

Reply:
Nice, LGTM.